### PR TITLE
Remove invalid comment in legacyregistry

### PR DESCRIPTION
**Pull request by sbs2001**
Signed-off-by: Shivam Sandbhor <shivam.sandbhor@gmail.com>

#### What type of PR is this?

/kind documentation

#### What this PR does / why we need it:

After the commit at https://github.com/kubernetes/kubernetes/commit/9ead9373f350c7ad438257a9e8b8977a67b900eb . The comment about the `Handler` function is no longer valid. This PR prevents misleading info in the docs.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/component-base/metrics/legacyregistry/registry.go [hunk 1]
```diff
@@ -53,9 +53,6 @@ func init() {
 // Handler returns an HTTP handler for the DefaultGatherer. It is
 // already instrumented with InstrumentHandler (using "prometheus" as handler
 // name).
-//
-// Deprecated: Please note the issues described in the doc comment of
```
Comment (dgrisonnet) [inline staging/src/k8s.io/component-base/metrics/legacyregistry/registry.go:57]: I am not sure to understand why the changes from https://github.com/kubernetes/kubernetes/commit/9ead9373f350c7ad438257a9e8b8977a67b900eb remove the deprecation from this function, could you please explain?

The issues mentioned in https://github.com/prometheus/client_golang/blob/v1.11.0/prometheus/promhttp/http.go#L216-L232 are still very valid to me.

Comment (dgrisonnet) [reply staging/src/k8s.io/component-base/metrics/legacyregistry/registry.go:57]: Yes after your explanation, it indeed makes a lot of sense to remove this comment. Thanks!

**Comment by sbs2001** [comment]
@dgrisonnet  I don't know why tests are failing at https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/106291/pull-kubernetes-verify/1460594597959831552. How can I fix this ?  Rather strange since this is just a change in a comment :)

**Comment by dgrisonnet** [comment]
You'll need to go to each place where a failure is mentioned and remove the linter directive that allows the use of a deprecated function. Such as https://github.com/kubernetes/kubernetes/blob/master/cmd/kube-proxy/app/server.go#L630-L631

**Comment by sbs2001** [comment]
@dgrisonnet Thanks. I did what you said, but now we need approval from 4 more sigs 😅

**Comment by dgrisonnet** [comment]
@sbs2001 could you please squash your commits?

**Comment by sbs2001** [comment]
@dgrisonnet  commits are squashed.

**Comment by dgrisonnet** [comment]
Since the changes from this PR are only lint-related, I don't think we will need approval from all the sig involved. Top-level approval should do.

/assign @wojtek-t
